trin-core/src/portalnet/storage.rs

```diff
@@ -693,6 +737,11 @@ const XOR_FIND_FARTHEST_QUERY: &str = "SELECT
 
 const TOTAL_DATA_SIZE_QUERY: &str = "SELECT TOTAL(content_size) FROM content_metadata";
 
+const TOTAL_ENTRY_COUNT_QUERY: &str = "SELECT COUNT(content_id_long) FROM content_metadata";
+
+const PAGINATE_QUERY: &str =
+    "SELECT content_id_long FROM content_metadata ORDER BY content_id_long DESC LIMIT :limit OFFSET :offset";
```

Comment:
I'm not sure if we want to order the `content_id` before setting the `offset`. Maybe I'm missing something but what is the use case for using the offset on ordered content ids?

Reply:
Hm, doesn't that open up to returning in random order and making it impossible to page through all the results, because you get random selections at each page?

Reply:
My thinking process is that the DB order of the records is the order they were created (the last saved content id is on the last row). Then. if we want to get, let's say the last 10 saved content keys, we can use an `offset = total_records_number - 10`. 

However, If we order the records by `content_id_long` (sha256 hash), this will sort the records in an order that is not predictable and then the last 10 records will be what I would call random.

Reply:
Ah, so you're proposing a new use case. The only previous use case mentioned was to enumerate *all* keys. The one your suggesting is to retrieve the most recent N keys added. So I guess the question I want to make sure we're asking is when would we use that feature? Is that just something that seems like a good idea, or are we ready to use it now?

Second, if we decide it's a good idea, SQL doesn't have guarantees about the default ordering when you don't provide an `ORDER BY` (and I'd rather not make assumptions about which SQL implementation we're using). So in order to reliably provide the feature, I think we'd need to add a new field. Something like a switching the primary key to an autoincrement or adding a new `create_timestamp` field to order by.

I'm not opposed to the idea, but I would like to see the specific example. Also, it feels like something to address in a follow-up PR.

Reply:
One more point to consider after speaking with @mrferris is whether or not we want to return all content keys, or all content keys for a specific subnetwork. Returning only the content keys belonging to a specific subnetwork will require an additional table column, but iiuc we need to do this anyway to support different radius / subnetwork. 

I don't have a strong opinion on whether we want to return content keys on a per-subnetwork basis, or simply enumerate them all. However, if we choose per-subnetwork, it's best to implement this asap so that we avoid the need to tear-down our testnet dbs down the road if we implement this feature later.

Reply:
> Ah, so you're proposing a new use case. The only previous use case mentioned was to enumerate _all_ keys. The one your suggesting is to retrieve the most recent N keys added. So I guess the question I want to make sure we're asking is when would we use that feature? Is that just something that seems like a good idea, or are we ready to use it now?

I don't think there are strongly defined use cases here. My initial understanding is that this RPC endpoint will be used for testing purposes (as Piper and Nick suggested) i.e to get visibility over the content keys stored in the network.
My intuition is that we may also want to know **when** content has been stored in a node, which can open the room for more detailed analyses.

I'm in favor of adding `timestamp` and `subnetwork` fields, it doesn't cost too much and I think this will give us better visibility over the content in the network.

Reply:
tracking this issue here: #480